save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/ContestEnroller.kt

```diff
@@ -114,11 +117,23 @@ fun ChildrenBuilder.showContestEnrollerModal(
     "ComplexMethod",
 )
 private fun contestEnrollerComponent() = FC<ContestEnrollerProps> { props ->
-    val (organizationName, setOrganizationName) = useState(props.organizationName)
-    val (projectName, setProjectName) = useState(props.projectName)
-    val (contestName, setContestName) = useState(props.contestName)
-    val (isContestSelector, _) = useState(props.contestName == null)
-    require(isContestSelector && props.organizationName != null && props.projectName != null || !isContestSelector && props.contestName != null)
+    val (isContestSelector, _) = useState(props.nameProps is ProjectNameProps)
```

Comment:
Now you can use `when` with smart casts instead of boolean flag and `as`

Reply:
We cannot do it with `var`